tools/do_meili_tools.py

```diff
@@ -1,23 +1,50 @@
 import time
 import socket
+import requests
 
 def wait_for_droplet_creation(droplet):
     while True:
         d = droplet.get_actions()
         if d[0].type == "create" and d[0].status == "completed":
-            droplet = droplet.load()
-            print("IP:", droplet.ip_address, "id:", droplet.id)
             break
         time.sleep(2)
 
 def wait_for_ssh_availability(droplet):
     while True:
+        time.sleep(2)
         s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         try:
             s.connect((droplet.ip_address, 22))
             s.shutdown(2)
             s.close()
-            break
+            return
         except Exception:
             continue
-        time.sleep(2)
\ No newline at end of file
+
+def wait_for_health_check(droplet):
+    while True:
+        time.sleep(2)
+        try:
+            resp = requests.get("http://{}/health".format(droplet.ip_address))
+            if resp.status_code >=200 and resp.status_code < 300:
+                return
+        except Exception as e:
+                continue
+
+def wait_for_droplet_shutdown(droplet):
+    while True:
+        try:
+            d = droplet.get_actions()
+            for act in d:
+                if act.type == "shutdown" and d[0].status == "completed":
+                    return
+        except Exception as e:
+            print("   Exception: {}".format(e))
+            return
+
+def wait_for_snapshot_creation(droplet):
+    while True:
+        time.sleep(2)
+        d = droplet.get_actions()
+        if d[0].type == "snapshot" and d[0].status == "completed":
+            return
```

Comment:
Why don't we catch the error here, like the function above?

Plus, could you add a new line at the end of the file please? Or fix #96 😇

Reply:
We can, indeed. I don't remember having any specific reason not to do it, maybe I never had an error on this and didn't realize it. But I'll add this (and fix #96 too) and re-request your review ;)